lib/poolex.ex

```diff
@@ -195,20 +189,29 @@ defmodule Poolex do
   """
   @spec run!(pool_id(), (worker :: pid() -> any()), list(run_option())) :: any()
   def run!(pool_id, fun, options \\ []) do
-    timeout = Keyword.get(options, :timeout, @default_wait_timeout)
-
-    {:ok, pid} = GenServer.call(pool_id, :get_idle_worker, timeout)
-
-    monitor_process = monitor_caller(pool_id, self(), pid)
+    checkout_timeout = Keyword.get(options, :checkout_timeout, @default_checkout_timeout)
+    {:ok, worker_pid} = get_idle_worker(pool_id, checkout_timeout)
+    monitor_process = monitor_caller(pool_id, self(), worker_pid)
```

Comment:
> Why do you need a process to monitor the current process, when you already have an after to release the busy worker anyway?

This monitoring helps to cover [this case](https://github.com/general-CbIC/poolex/blob/develop/test/poolex_test.exs#L257-L264).

If the caller was killed "outside" while working with the received worker, then without monitoring, Poolex will not know about it.

At first, I thought about doing regular monitoring, but I didn't want to complicate the storage of links in ETS and add metadata for links of callers and workers there.

Then I thought about making a two-way link via `Process.link()`, but I don't want the caller to die if the worker suddenly dies. This is incorrect behavior.

Therefore, I decided to bind in one direction: if the caller dies, we process this event and kill the worker.

Reply:
`:poolboy` monitors the processes which have taken the workers. It does because it has the `checkout/checkin` interface, which is fundamental if you want to deal with workers in a GenServer. I arrived at this point when I realized it was too much work for me.

If you don't add the `checkout/checkin` interface, *maybe* the after it is enough, but I'm not 100% sure. I would monitor from `:poolex`.

Reply:
> `:poolboy` monitors the processes which have taken the workers

To solve such a problem, spawning monitoring is cheaper than storing metadata in ETS or state. But correcting this over time is possible if there is a beautiful solution.

> It does because it has the checkout/checkin interface, which is fundamental if you want to deal with workers in a GenServer.

`Poolex` has `get_idle_worker/2` instead of `checkout` and `cast :release_busy_worker` instead of `checkin`.